### ISSUE TITLE
Execution parameters of the paper

### ISSUE THREAD

**Issue by juarezsacenti** [TaoMiner/joint-kg-recommender#9]
Dear TaoMiner,

My name is Juarez A. P. Sacenti and I am a Ph.D. candidate at Federal University of Santa Catarina, Brazil. My thesis is about knowledge-based recommender systems and I would like to conduct a comparative analysis of different knowledge bases as input of your work using your github repository (https://github.com/TaoMiner/joint-kg-recommender). 

I would like to know whether you would share with us the execution parameters of evaluated algorithms (FM, BPRMF, CFKG, CKE, CoFM(reg), CoFM(share), TUP(hard), TUP(soft), KTUP(hard), KTUP(soft)) in your publication "Unifying Knowledge Graph Learning and Recommendation: Towards a Better Understanding of User Preferences", specifically for MovieLens dataset (ml1m).

I look forward to hearing from you.

**Comment by MathematicalModels** [comment]
> Dear TaoMiner,
> 
> My name is Juarez A. P. Sacenti and I am a Ph.D. candidate at Federal University of Santa Catarina, Brazil. My thesis is about knowledge-based recommender systems and I would like to conduct a comparative analysis of different knowledge bases as input of your work using your github repository (https://github.com/TaoMiner/joint-kg-recommender).
> 
> I would like to know whether you would share with us the execution parameters of evaluated algorithms (FM, BPRMF, CFKG, CKE, CoFM(reg), CoFM(share), TUP(hard), TUP(soft), KTUP(hard), KTUP(soft)) in your publication "Unifying Knowledge Graph Learning and Recommendation: Towards a Better Understanding of User Preferences", specifically for MovieLens dataset (ml1m).
> 
> I look forward to hearing from you.

Dear Sacenti, 

Have you received their sharing of the parameters?

Best Regards

**Comment by auhjin** [comment]
unsubscribe.













At 2021-03-25 11:24:35, "MathematicalModels" ***@***.***> wrote:

Dear TaoMiner,

My name is Juarez A. P. Sacenti and I am a Ph.D. candidate at Federal University of Santa Catarina, Brazil. My thesis is about knowledge-based recommender systems and I would like to conduct a comparative analysis of different knowledge bases as input of your work using your github repository (https://github.com/TaoMiner/joint-kg-recommender).

I would like to know whether you would share with us the execution parameters of evaluated algorithms (FM, BPRMF, CFKG, CKE, CoFM(reg), CoFM(share), TUP(hard), TUP(soft), KTUP(hard), KTUP(soft)) in your publication "Unifying Knowledge Graph Learning and Recommendation: Towards a Better Understanding of User Preferences", specifically for MovieLens dataset (ml1m).

I look forward to hearing from you.

Dear Sacenti,

Have you received their sharing of the parameters?

Best Regards

**Comment by MathematicalModels** [comment]
> unsubscribe. At 2021-03-25 11:24:35, "MathematicalModels" ***@***.***> wrote: Dear TaoMiner, My name is Juarez A. P. Sacenti and I am a Ph.D. candidate at Federal University of Santa Catarina, Brazil. My thesis is about knowledge-based recommender systems and I would like to conduct a comparative analysis of different knowledge bases as input of your work using your github repository (https://github.com/TaoMiner/joint-kg-recommender). I would like to know whether you would share with us the execution parameters of evaluated algorithms (FM, BPRMF, CFKG, CKE, CoFM(reg), CoFM(share), TUP(hard), TUP(soft), KTUP(hard), KTUP(soft)) in your publication "Unifying Knowledge Graph Learning and Recommendation: Towards a Better Understanding of User Preferences", specifically for MovieLens dataset (ml1m). I look forward to hearing from you. Dear Sacenti, Have you received their sharing of the parameters? Best Regards — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe.

Excuse me?

**Comment by juarezsacenti** [comment]
Dear @MathematicalModels,

Unfortunately, I did not receive any answer from @TaoMiner, either by github or by email.
After reading carefully their paper, I adopted the following parameters for Movielens 1M (ml1m):
| Algorithm | Pretrained |  Learning rate | L_2 | Optimizer | Batch size | Embedding size | Jointly rate | Eval interval steps | Training steps | Early stop steps to wait |
|----------|:-------------:|:-------------:|:-------------:|:-------------:|:-------------:|:-------------:|:-------------:|:-------------:|:-------------:|------:|
| TransE | - | 0.001 | 0 | Adam | 256 | 100 | - | 9150 | 915000 | 45750 |
| TransH | TransE | 0.001 | 0 | Adam | 256 | 100 | - | 9150 | 915000 | 45750 |
| BPRMF | - | 0.005 | 10^-5 | Adagrad | 256 | 100 | - | 5000 | 500000 | 25000 |
| TUP(soft) | BPRMF | 0.005 | 10^-5 | Adagrad | 256 | 100 | - | 500 | 50000 | 2500 |
| CFKG | BPRMF:TransE | 0.005 | 0 | Adam | 256 | 100 | 0.5 | 19520 | 1952000 | 97600 |
| CKE | BPRMF:TransE | 0.005 | 0 | Adam | 256 | 100 | 0.5 | 19520 | 1952000 | 97600 |
| CoFM(reg) | BPRMF:TransE | 0.005 | 0 | Adam | 256 | 100 | 0.5 | 19520 | 1952000 | 97600 |
| KTUP(soft) | TUP:TransH | 0.005 | 0 | Adam | 256 | 100 | 0.5 | 19520 | 1952000 | 97600 |

Also, I keep "-L1_flag -norm_lambda 1 -kg_lambda 1" flags.
Note that these settings do not achieved the same results of their paper and I do not intend to obtain the optimal result.

Maybe @auhjin meant that @TaoMiner have unsubscribed from this tread. Am I right?

Best Regards

**Comment by MathematicalModels** [comment]
> Dear @MathematicalModels,
> 
> Unfortunately, I did not receive any answer from @TaoMiner, either by github or by email.
> After reading carefully their paper, I adopted the following parameters for Movielens 1M (ml1m):
> 
> Algorithm	Pretrained	Learning rate	L_2	Optimizer	Batch size	Embedding size	Jointly rate	Eval interval steps	Training steps	Early stop steps to wait
> TransE	-	0.001	0	Adam	256	100	-	9150	915000	45750
> TransH	TransE	0.001	0	Adam	256	100	-	9150	915000	45750
> BPRMF	-	0.005	10^-5	Adagrad	256	100	-	5000	500000	25000
> TUP(soft)	BPRMF	0.005	10^-5	Adagrad	256	100	-	500	50000	2500
> CFKG	BPRMF:TransE	0.005	0	Adam	256	100	0.5	19520	1952000	97600
> CKE	BPRMF:TransE	0.005	0	Adam	256	100	0.5	19520	1952000	97600
> CoFM(reg)	BPRMF:TransE	0.005	0	Adam	256	100	0.5	19520	1952000	97600
> KTUP(soft)	TUP:TransH	0.005	0	Adam	256	100	0.5	19520	1952000	97600
> Also, I keep "-L1_flag -norm_lambda 1 -kg_lambda 1" flags.
> Note that these settings do not achieved the same results of their paper and I do not intend to obtain the optimal result.
> 
> Maybe @auhjin meant that @TaoMiner have unsubscribed from this tread. Am I right?
> 
> Best Regards

Dear Sacenti,
         Thank you so much for your generous sharing!

Best Regards

**Comment by juarezsacenti** [comment]
Dear @MathematicalModels,

In the last weeks I received from the authors the attached parameters.

[parameters.txt](https://github.com/TaoMiner/joint-kg-recommender/files/6361641/parameters.txt)

I reproduce the experiments with ml dataset, obtaining the following results:

  | CFKG | CKE | CoFM | KTUP(hard) | KTUP(soft)
-- | -- | -- | -- | -- | --
p@10 | 0.2803 | 0.3879 | 0.3212 | 0.4023 | 0.4101
r@10 | 0.1165 | 0.1648 | 0.1292 | 0.1704 | 0.1723
ndcg | 0.5778 | 0.6698 | 0.5855 | 0.6897 | 0.6994

Best Regards

**Comment by MathematicalModels** [comment]
> Dear @MathematicalModels,
> 
> In the last weeks I received from the authors the attached parameters.
> 
> [parameters.txt](https://github.com/TaoMiner/joint-kg-recommender/files/6361641/parameters.txt)
> 
> I reproduce the experiments with ml dataset, obtaining the following results:
> 
>  	CFKG	CKE	CoFM	KTUP(hard)	KTUP(soft)
> p@10	0.2803	0.3879	0.3212	0.4023	0.4101
> r@10	0.1165	0.1648	0.1292	0.1704	0.1723
> ndcg	0.5778	0.6698	0.5855	0.6897	0.6994
> Best Regards

Thank you very much for the kind sharing ：）
